File: API/Helpers/UserParams.cs
namespace API.Helpers
{
    public class UserParams:PaginationParams
    {
        
        public string? CurrentUsername { get; set; }
        public string? Gender { get; set; }
        public int MinAge { get; set; }
        public int MaxAge { get; set; }

        public string orderBy { get; set; } = "lastActive";
    }
}

File: client/src/app/_models/userParams.ts
import { User } from "./User";

export class userParams {
    pageNumber = 1;
    pageSize = 10;
    gender: string;
    minAge = 18;
    maxAge = 100;
    orderBy='lastActive';

    constructor(user: User) {

        this.gender = user.gender === 'male' ? 'female' : 'male';

    }

}